ml-agents/mlagents/trainers/saver/saver.py

```diff
@@ -0,0 +1,65 @@
+# # Unity ML-Agents Toolkit
+import abc
+from typing import Any
+
+
+class BaseSaver(abc.ABC):
+    """This class is the base class for the Saver"""
+
+    def __init__(self):
+        pass
+
+    def register(self, module: Any) -> None:
+        """
+        Register the modules to the Saver.
+        The Saver will store the module and include it in the saved files
+        when saving checkpoint/exporting graph.
+        :param module: the module to be registered
+        """
+        pass
+
+    def _register_policy(self, policy):
```

Comment:
Should we add typing to policy and optimizer for these two (I'm guessing the generic `Policy` and `Optimizer`).

Reply:
definition in tf_saver will have typing conflicts with super class